### DISCUSSION TITLE
best way chain processes with yazi's API ?

### DISCUSSION THREAD

**Discussion by teto** [sxyazi/yazi#1103] [1. Q&A]
I created a plugin to preview `compressed json` files (compressed with zlib).

Right now I just decompress the file:

```
  ya.preview_widgets(self, { ui.Paragraph.parse(self.area, "Loading jsonzlib...") })
	local child = Command("pigz") -- decompress the file https://zlib.net/pigz/
		:args({
          '-dc',
			tostring(self.file.url),
		})
		:stdout(Command.PIPED)
		:stderr(Command.PIPED)
		:spawn()
```

but I would like to prettify the resulting json afterwards as well. And I wonder what is the best way to do it ?

Can I expect Command to run in a shell and do something like ?
```
	local child = Command("pigz -dc "..tostring(self.file.url).." | jq '.' ") 
		:spawn()
```
or should I pipe the first output as input of a second `Command` ? 

I thought about creating an intermediary script but I am curious about a cleaner way

**Comment by sxyazi** [comment]
Yeah it's possible, see https://github.com/sxyazi/yazi/pull/1033

It currently requires the nightly version of Yazi
